### PR TITLE
MINIFICPP-1288 - FlowController C2 init refactor

### PR DESCRIPTION
**Pull request by hunyadi-dev**
Split up FlowController::initializeC2, remove atomicity from c2_initialized_, delete flow_update_ field.

### CODE REVIEW

File: libminifi/src/FlowController.cpp [hunk 1]
```diff
@@ -216,9 +215,6 @@ bool FlowController::applyConfiguration(const std::string &source, const std::st
   bool started = false;
   try {
     load(this->root_, true);
-    flow_update_ = true;
-    started = start() == 0;
```
Comment (lordgamez) [inline libminifi/src/FlowController.cpp:220]: Was this supposed to be removed? The code in the conditional branch `if (started) {` below now seems to be dead code.

File: libminifi/src/FlowController.cpp [hunk 2]
```diff
@@ -441,43 +429,40 @@ void FlowController::initializeC2() {
...
+  return agent_identifier;
+}
 
-  if (c2_initialized_ && !flow_update_) {
-    return;
+void FlowController::addQueueMetricsToDeviceInformation() {
+  std::shared_ptr<state::response::QueueMetrics> queueMetrics = std::make_shared<state::response::QueueMetrics>();
+  std::map<std::string, std::shared_ptr<Connection>> connections;
+  root_->getConnections(connections);
+  for (auto con : connections) {
```
Comment (lordgamez) [inline libminifi/src/FlowController.cpp:451]: This could be `const auto&`

File: libminifi/src/FlowController.cpp (hunk 1 above)
Comment (hunyadi-dev) [reply libminifi/src/FlowController.cpp:220]: Honestly, I don't remember what was exactly extracted into this PR. I think this was opened to ease the review load on #890, but the boundaries are probably not correctly set. Closing this PR.

**Comment by hunyadi-dev** [comment]
Probably part of #890, not worth rebasing separately.

**Pull request closed without merging**